### PR TITLE
[ASTextNode] Fix highlighting when textContainerInsets are set.

### PR DESCRIPTION
**Pull request by nlutsenko**
Since `textContainerInset` only impact the calculated size, and not glyph sizes - we need to add the insets to all calculated pieces.

### CODE REVIEW

**Comment by nlutsenko** [comment]
Before:
simulator screen shot aug 15 2016 1 31 12 pm

After:
simulator screen shot aug 15 2016 1 30 57 pm

**Comment by Adlai-Holler** [comment]
Awesome catch @nlutsenko ! Would you be willing to add a snapshot test to ASTextNodeSnapshotTests.m to ensure we don't regress on this?

**Comment by nlutsenko** [comment]
Let me try doing that.
Not sure if I can get it fast though, so if I get blocked - I am going to say 'no' :grin:

**Comment by Adlai-Holler** [comment]
Lol fair enough!

**Comment by nlutsenko** [comment]
Ok, @Adlai-Holler, snapshot test case added.
It's slightly hacky, compared to a general one, since the highlight layer is actually getting rendered via a special layer, but it works and tests the proper thing.

Also, I noticed that the whole tests file is missing from the project, so added it as well.
Let me know how it looks...

File: AsyncDisplayKitTests/ASTextNodeSnapshotTests.m [hunk 1]
```diff
@@ -32,4 +32,26 @@ - (void)testTextContainerInset
...
+  textNode.frame = CGRectMake(50, 50, textNode.calculatedSize.width, textNode.calculatedSize.height);
+  textNode.textContainerInset = UIEdgeInsetsMake(5, 10, 10, 5);
+
+  [backgroundView addSubview:textNode.view];
+  backgroundView.frame = UIEdgeInsetsInsetRect(textNode.bounds, UIEdgeInsetsMake(-50, -50, -50, -50));
+
+  textNode.highlightRange = NSMakeRange(0, textNode.attributedText.length);
+
+  [ASSnapshotTestCase hackilySynchronouslyRecursivelyRenderNode:textNode];
+  FBSnapshotVerifyLayer(backgroundView.layer, nil);
```
Comment (Adlai-Holler) [inline AsyncDisplayKitTests/ASTextNodeSnapshotTests.m]: Awesome! Can these two lines be replaced by `ASSnapshotVerifyNode(textNode, nil)`?

Comment (nlutsenko) [inline AsyncDisplayKitTests/ASTextNodeSnapshotTests.m]: Nope. I need the superview/superlayer.

**Comment by Adlai-Holler** [comment]
It looks awesome! Thank you I'm cool with merging this as soon as @maicki reviews and the build passes.

**Comment by nlutsenko** [comment]
I believe the tests failing are completely unrelated to the change made...
@maicki, is that what you are fixing with #2076?

**Comment by maicki** [comment]
@nlutsenko Yes that's what I'm fixing with the PR. Sorry about that ...

**Comment by nlutsenko** [comment]
All good, will track it and rebase this one after it's merged.

**Comment by nlutsenko** [comment]
Rebased on top of latest master, with your change merged.
Let's see if it passes now.

**Comment by hannahmbanana** [comment]
Thanks @nlutsenko! ASTextNode bug fixes are always extra appreciated. :) Are you working with the Instant Articles team?

**Comment by nlutsenko** [comment]
Yup.

**Pull request merged**